inst/include/sampler.h

```diff
@@ -551,6 +548,21 @@ namespace smc {
         }
     }
     
+	template <class Space>
+	void sampler<Space>::Multinomial(unsigned n, unsigned k, arma::vec w, unsigned int * X) {		
```

Comment:
You can't (really) cast unsigned* to int* because there's no guarantee that int and unsigned int have a closely related interpretation. You could use reinterpret_cast to tell the compiler to pretend that an unsigned* is an int* which would produce code that compiles... but it would have undefined behaviour.

The only reason for using an unsigned type was that it was a better logical description of a number of replicates. If there are pragmatic advantages for using a signed type, as there seem to be, then indeed that seems sensible.

(Don't feel that you have to respond to every question with new code, by the way, I was just curious as to whether it was really necessary -- and as Dirk said, there's no reason that small things like this can't be tidied up later if it's desirable to.)